src/v2/ion/uiSchemaTransformer.js

```diff
@@ -29,6 +29,16 @@
  * @param {IONFormField[]} remediationValue
  */
 const createUISchema = (remediationValue = []) => {
+  // For casee where field itself is a form, it has a formname and we are appending the formname to each field
+  // This is so that while making the request we can bundle these key:value pairs under the same key name
+  // For simplicity we are assuming that when field itself is a form its only one level deep
+  if (remediationValue[0] && remediationValue[0].name && remediationValue[0].form) {
```

Comment:
why you choose index `0`? what if the 2nd item is has `form` attribute?

```
{
    "value": [
        {
            "name": "Password"
        },
        {
            "name": "userProfile",
            "form": {
                "value": [
                    {
                        "name": "lastName",
                        "label": "Last name",
                        "required": true
                    },
                ]
            }
        }
    ]
}
```

I think the we need loop through the `remediationValue` to convert each entry that has `form`. something like (psudecode)
```
newValue = _.chain(remediationValue)
  .map(v => {
    if (v.form) {
        const inputGroupName = v.name;
        return v.form.value.map(input => {
            input.name = inputGroupName + '.' + input.name;
            return input;
          });
    } else {
        return v
    }
  })
  .flatten()
  .value();
```

Reply:
This was mostly because we currently only are supporting enrollment of userProfile.